projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -404,6 +411,98 @@ private function parse_menu_item( $title ) {
 
 		return $item;
 	}
+
+	/**
+	 * Check if the given string is structured like a domain
+	 *
+	 * @TODO: Support deep nested domains like jp.eu.foo.something.something.wordpress.com
+	 * @param string $domain_name_value The string that will be checked.
+	 *
+	 * @return false|int
+	 */
+	private function is_valid_domain_name( $domain_name_value = '' ) {
+		return preg_match( '/^(http[s]?\:\/\/)?((\w+)\.)?(([\w-]+)?)(\.[\w-]+){1,2}$/', $domain_name_value );
+	}
+
+	/**
+	 * Generate a identifier based on menu or submenu slug
+	 *
+	 * @param string $key Menu or submenu slug.
+	 *
+	 * @return string
+	 */
+	private function generate_identifier( $key ) {
+		$replace_table = array(
+			'http://'  => '',
+			'https://' => '',
+			'-'        => '_',
+			'.php'     => '', // we add this in order to avoid cases where customizer.php can be considered a domain.
+		);
+
+		// Split string on '/' or '?' to account for query strings.
+		$array_of_paths = preg_split( '/(\/|\?)/', strtr( $key, $replace_table ), -1, PREG_SPLIT_NO_EMPTY );
+
+		// Dont touch domain names as these get filtered out next and replace non-alphanumeric characters with an underscore.
+		$parse_query_strings = $this->format_non_domain_identifiers( $array_of_paths );
+
+		$array_without_domains = array_filter(
+			$parse_query_strings,
+			function ( $value ) {
+				return ! $this->is_valid_domain_name( $value );
+			}
+		);
+
+		return sanitize_key( implode( '_', $array_without_domains ) );
+	}
+
+	/**
+	 * Format non-domain identifiers
```

Comment:
Could we add more description as to _how_ and what kind of "formatting" is being applied?